nixos/modules/services/misc/nix-optimise.nix

```diff
@@ -0,0 +1,49 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.nix.optimise;
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    nix.optimise = {
+
+      automatic = mkOption {
+        default = false;
+        type = types.bool;
+        description = "Automatically run the nix store optimiser at a specific time.";
+      };
+
+      dates = mkOption {
+        default = "03:45";
+        type = types.str;
```

Comment:
NTS: that's as of commit 488f0d9cb362342081ece69e6ef52cd4adf4fa5a, which isn't in `nixos-unstable` yet.